2022/codec_registry_shivank/README.md

```diff
@@ -1 +1,93 @@
+## Progress and updates for the project [Building Codec Registry](https://summerofcode.withgoogle.com/programs/2022/projects/g4IPN5HL) by [Shivank Chaudhary](https://github.com/alt-shivam) for Google Summer of Code 2022
+
+## About Me
+
+Name: Shivank Chaudhary
+Email: shivank1128@gmail.com
+
+University: Ramanujan College, University of Delhi, Kalkaji, New Delhi (IN)  
+Field of Study: Bachelors in Computer Science (Hons.)  
+Grade: 9.18 CGPA  
+GitHub: [Alt-Shivam](https://github.com/Alt-Shivam)  
+LinkedIn: https://www.linkedin.com/in/shivank-chaudhary-a67681202/  
+
+Location: Noida, Uttar Pradesh, IN (201301)
+
+Time Zone: IST( UTC + 5:30)
+
+## Project Information
+Project Idea: Zarr is a format for compressed N-Dimensional arrays and uses many compression algorithms to handle the data. Each of these compression algorithms is bound with an identifier to identify them, But we don’t have any registry to help other languages identify them. To fill this gap we need a registry to be defined for these codecs which is human-readable as well as machine-readable.  
+
+**Project length:** 175 hrs
+
+## Related links:
+1) https://github.com/zarr-developers/numcodecs
+2) https://github.com/zarr-developers/numcodecs/issues/278
+3) https://support.hdfgroup.org/services/filters.html
+4) https://github.com/cgohlke/imagecodecs
+
+# Project Weekly Updates:
+
+## Pre-GSoC Work
+* Created a Research file.
+* Documented several codecs into this format.
+```
+* Blosc
+- A brief about how the compression technique works(2-3 lines)
+- Where the codec is primarily used(webpages, PNGs, JPEGs, OS etc.)
+- Comparing compression and decompression benchmarks/speeds
+- Any other info you think is useful
+- Important links for the codec(GitHub, Blog post, Implementations etc.)
+```
+* Created an initial version of Research file on Codecs.
+* Created a Pull Request for [Pre-commit configuration](https://github.com/zarr-developers/zarr-python/pull/1015)
+
+## Week 1 & 2
+* Added following new codecs.
+```
+PackBits
+Pickle
+Quantize
+Shuffle
+ZFPY
+Astype
+Bitround
+JSON
+```
+* Documented Codec informations in Research file.
+* Created a Pull Request [Link](https://github.com/zarr-developers/zarr-python/pull/1016)
+
+## Week 3 & 4
+* Created a Github Repo to keep registry files. [Link](https://github.com/Alt-Shivam/Codecs-Registry)
+* Created a **csv** file for Registry.
+* Added ImageCodecs.
+```
+AEC
+APNG
+AVIF
+Brotli
+Brunsli
+```
+and there documentation in research file.
+* Created a Pull Request. [Link](https://github.com/zarr-developers/zarr-python/pull/1047)
+
+## Week 5 & 6
+* Created a Github-Pages website to show Codec Registry. [Link](https://alt-shivam.github.io/Codecs-Registry/)
+* Shifted Codec research file from docs to .md.
+* Created a saperate page to show the research file. [Link](https://alt-shivam.github.io/Codecs-Registry/Others/Research.html)
+* Added Basic Information on CodecRegistry website.
+
+# Mid-Term Evaluations. :tada:
```

Comment:
If you want to add something here to show that you've passed the mid-term evaluation, that would be great. Maybe a screenshot from GSoC 2022 portal?

Reply:
Great Idea, Thanks